src/lxc/start.c

```diff
@@ -900,6 +885,34 @@ static int do_start(void *data)
 		goto out_warn_father;
 	}
 
+	/* The container has been setup. We can now switch to an unprivileged
+	 * uid/gid.
+	 */
+	if (handler->conf->is_execute) {
+		uid_t new_uid = 0;
+		gid_t new_gid = 0;
+
+		if (handler->conf->init_uid > 0)
```

Comment:
uid_t is (according to the headers I'm looking at) unsigned int, so I think you can just do

uid_t new_uid = handler->conf->init_uid;
gid_t new_gid = handler->conf->init_gd;

saving a few lines.